servers/rendering/shader_language.h

```diff
@@ -769,7 +780,8 @@ class ShaderLanguage {
 		Map<StringName, BuiltInInfo> built_ins;
 		Map<StringName, StageFunctionInfo> stage_functions;
 
-		bool can_discard;
+		bool can_discard = false;
+		bool main_function = false;
```

Comment:
The concept of "main function" is a little confusing for me. Why are "fragment", "vertex", and "light" "main functions", but "process" and "sky" are not?

This flag seems to flag functions that can pass data to other functions. So maybe a different name would make that clear?

Reply:
They should be, unless I messed up, need to give it a check.

Reply:
Main function is just a hint to the compiler, mostly that it should not return anything. SubFunctions don't get this set.